src/transformers/models/t5/modeling_t5.py

```diff
@@ -776,9 +780,14 @@ def _shift_right(self, input_ids):
         ), "self.model.config.decoder_start_token_id has to be defined. In T5 it is usually set to the pad_token_id. See T5 docs for more information"
 
         # shift inputs to the right
-        shifted_input_ids = input_ids.new_zeros(input_ids.shape)
-        shifted_input_ids[..., 1:] = input_ids[..., :-1].clone()
-        shifted_input_ids[..., 0] = decoder_start_token_id
+        if is_torch_fx_available() and isinstance(input_ids, torch.fx.Proxy):
```

Comment:
If we are going to use this combo a lot, then down the road we could consider combining these 2 with a helper, so we won't need to repeat the code:
```
if is_torch_fx_proxy(input_ids): ...
```